File: dart/abstract_factory/sport_brand/bin/product.dart
abstract class Shoe {
  Map<String, dynamic> generate(double size);
}

class AdidasShoe implements Shoe {
  @override
  Map<String, dynamic> generate(double size) {
    Map<String, dynamic> result = Map();
    result['logo'] = 'Adidas';
    result['size'] = size;
    return result;
  }
}

class NikeShoe implements Shoe {
  @override
  Map<String, dynamic> generate(double size) {
    Map<String, dynamic> result = Map();
    result['logo'] = 'Nike';
    result['size'] = size;
    return result;
  }
}

abstract class Shirt {
  Map<String, String> generate(String size);
}

class AdidasShirt implements Shirt {
  @override
  Map<String, String> generate(String size) {
    Map<String, String> result = Map();
    result['logo'] = 'Adidas';
    result['size'] = size;
    return result;
  }
}

class NikeShirt implements Shirt {
  @override
  Map<String, String> generate(String size) {
    Map<String, String> result = Map();
    result['logo'] = 'Nike';
    result['size'] = size;
    return result;
  }
}

File: typescript/abstract_factory/sport_brand/product.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NikeShirt = exports.AdidasShirt = exports.NikeShoe = exports.AdidasShoe = void 0;
class AdidasShoe {
    generate(size) {
        const result = new Map();
        result.set('logo', 'Adidas');
        result.set('size', size);
        return result;
    }
}
exports.AdidasShoe = AdidasShoe;
class NikeShoe {
    generate(size) {
        const result = new Map();
        result.set('logo', 'Nike');
        result.set('size', size);
        return result;
    }
}
exports.NikeShoe = NikeShoe;
class AdidasShirt {
    generate(size) {
        const result = new Map();
        result.set('logo', 'Adidas');
        result.set('size', size);
        return result;
    }
}
exports.AdidasShirt = AdidasShirt;
class NikeShirt {
    generate(size) {
        const result = new Map();
        result.set('logo', 'Nike');
        result.set('size', size);
        return result;
    }
}
exports.NikeShirt = NikeShirt;
